Automattic/newspack-plugin
# feat(ras): enable custom contact metadata prefixes

### All Submissions:






### Changes proposed in this Pull Request:



Adds a field to the RAS settings page to set the prefix for contact metadata field keys. Defaults to `NP_` which ensures continuity on sites already using RAS.

<img width="1081" alt="Screen Shot 2023-01-24 at 4 38 43 PM" src="https://user-images.githubusercontent.com/2230142/214445181-e226c011-bb2f-4ee8-b7c7-50b7374ca91c.png">

Closes 1200550061930446/1203749513702559.

### How to test the changes in this Pull Request:

1. On a site with RAS enabled and connected to ActiveCampaign, visit the **Newspack > Engagement > Reader Activation** dashboard. Confirm there's a new "Metadata Field Prefix" field if the the "Synchronize reader to ESP" is enabled, and that it's set to `NP_`.
2. Without making changes, smoke test the field syncing to ensure that it still works with the default behavior: make a donation and ensure that the contact is created in ActiveCampaign with all the Newspack fields prefixed with `NP_`.
3. Update the "Metadata Field Prefix" field to a new string and save.
4. Make a new donation using the same email address as in step 2. Confirm in AC that the contact now has a new set of fields with the prefix you set in step 3 which contain the most updated data, in addition to the `NP_` fields from before. (The old fields will remain to prevent data loss, but will no longer be synced or updated unless you reset the prefix.)
5. Make a new donation in a new session with a different email address. Confirm in AC that the data for this contact is only synced to the fields with the new prefix.

* Note: The new fields will appear in the General Details field group in ActiveCampaign. They can be moved to a different group in the AC dashboard.

### Other information:
